comp/otelcol/collector/pipeline.go

```diff
@@ -0,0 +1,97 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package collector
+
+import (
+	"context"
+
+	"go.uber.org/fx"
+
+	"github.com/DataDog/datadog-agent/comp/core/config"
+	corelog "github.com/DataDog/datadog-agent/comp/core/log"
+	logsagent "github.com/DataDog/datadog-agent/comp/logs/agent"
+	"github.com/DataDog/datadog-agent/pkg/logs/message"
+	"github.com/DataDog/datadog-agent/pkg/metadata/inventories"
+	"github.com/DataDog/datadog-agent/pkg/otlp"
+	"github.com/DataDog/datadog-agent/pkg/serializer"
+	"github.com/DataDog/datadog-agent/pkg/util"
+)
+
+// dependencies specifies a list of dependencies required for the collector
+// to be instantiated.
+type dependencies struct {
+	fx.In
+
+	// Lc specifies the fx lifecycle settings, used for appending startup
+	// and shutdown hooks.
+	Lc fx.Lifecycle
+
+	// Config specifies the Datadog Agent's configuration component.
+	Config config.Component
+
+	// Log specifies the logging component.
+	Log corelog.Component
+
+	// Serializer specifies the metrics serializer that is used to export metrics
+	// to Datadog.
+	Serializer serializer.MetricSerializer
+
+	// LogsAgent specifies a logs agent
+	LogsAgent util.Optional[logsagent.Component]
+}
+
+type collector struct {
+	deps dependencies
+	col  *otlp.Pipeline
+}
+
+func (c *collector) Start() error {
```

Comment:
What is the reason of having `Start` and `Stop` methods instead of using [lifecycle-hooks](https://uber-go.github.io/fx/lifecycle.html#lifecycle-hooks)?